desktop/cmp/panel/Panel.ts

```diff
@@ -21,13 +22,71 @@ import {useContextMenu, useHotkeys} from '@xh/hoist/desktop/hooks';
 import '@xh/hoist/desktop/register';
 import {splitLayoutProps} from '@xh/hoist/utils/react';
 import {castArray, omitBy} from 'lodash';
-import PT from 'prop-types';
-import {Children, isValidElement, useRef} from 'react';
+import {Children, isValidElement, ReactElement, ReactNode, useRef} from 'react';
 import {modalSupport} from '../modalsupport/ModalSupport';
 import {panelHeader} from './impl/PanelHeader';
 import {resizeContainer} from './impl/ResizeContainer';
 import './Panel.scss';
 import {PanelModel} from './PanelModel';
+import {HotkeyConfig} from '@xh/hoist/kit/blueprint';
+
+
+interface PanelProps extends BoxProps<PanelModel> {
+    /** True to style panel header (if displayed) with reduced padding and font-size. */
+    compactHeader?: boolean,
+
+    /** Items to be added to the right-side of the panel's header. */
+    headerItems?: ReactNode[],
+
+    /** An icon placed at the left-side of the panel's header. */
+    icon?: ReactElement,
+
+    /**
+     * Specification of a context menu.
+     *  @see useContextMenu() for more information on accepted values for this prop.
+     */
+    contextMenu?: any;
```

Comment:
Should probably leave a todo here to remind us to replace this with `ContextMenuConfig` when its ready

Reply:
👍🏻